registries/adapters/helm_adapter.go

```diff
@@ -0,0 +1,326 @@
+	var imageNames []string
+
+	r.Charts = map[string]ChartVersions{}
+
+	index, err := r.getHelmIndex()
+	if err != nil {
+		return imageNames, err
+	}
+
+	for name, entry := range index.Entries {
+		if len(entry) == 0 {
+			continue
+		}
+
+		r.Charts[name] = entry
+		imageNames = append(imageNames, name)
+	}
+
+	return imageNames, nil
+}
+
+// FetchSpecs - retrieve the spec for the image names.
+func (r *HelmAdapter) FetchSpecs(imageNames []string) ([]*apb.Spec, error) {
+	var specs []*apb.Spec
+
+	for _, name := range imageNames {
+		var (
+			chartVersions []string
+			values        string
+		)
+
+		charts, ok := r.Charts[name]
+		if !ok {
+			continue
+		}
+
+		for _, chart := range charts {
+			chartVersions = append(chartVersions, chart.Version)
+		}
+		// Use the latest chart for creating the bundle
+		chart := charts[0]
+
+		if len(chart.URLs) > 0 {
+			resp, err := http.Get(chart.URLs[0])
+			if err != nil {
+				continue
+			}
+			defer resp.Body.Close()
+
+			values = r.loadArchive(resp.Body)
+		}
+
+		// Convert chart to Bundle Spec
+		spec := &apb.Spec{
+			Runtime:     2,
+			Version:     "1.0",
+			Async:       "optional",
+			Bindable:    false,
+			Image:       r.Config.Runner,
+			FQName:      chart.Name,
+			Tags:        chart.Keywords,
+			Description: chart.Description,
+			Metadata: map[string]interface{}{
+				//"longDescription":  chart.Description,
+				"displayName":      fmt.Sprintf("%s (Helm)", chart.Name),
+				"documentationUrl": chart.Home,
+				"dependencies":     chart.Sources,
+				"imageUrl":         chart.Icon,
+			},
+			Plans: []apb.Plan{
+				apb.Plan{
+					Name:        "default",
+					Description: "Default plan for running helm charts",
+					Parameters: []apb.ParameterDescriptor{
+						apb.ParameterDescriptor{
+							Name:      "repo",
+							Title:     "Helm Chart Repository URL",
+							Type:      "string",
+							Default:   r.Config.URL.String(),
+							Pattern:   fmt.Sprintf("^%s$", r.Config.URL.String()),
+							Updatable: false,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:      "chart",
+							Title:     "Helm Chart",
+							Type:      "string",
+							Default:   chart.Name,
+							Pattern:   fmt.Sprintf("^%s$", chart.Name),
+							Updatable: false,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:      "version",
+							Title:     "Helm Chart Version",
+							Type:      "enum",
+							Enum:      chartVersions,
+							Default:   chart.Version,
+							Updatable: true,
+							Required:  false,
+						},
+						apb.ParameterDescriptor{
+							Name:        "values",
+							Title:       "Values",
+							Type:        "string",
+							DisplayType: "textarea",
+							Default:     values,
+							Updatable:   true,
+							Required:    false,
+						},
+					},
+				},
+			},
+		}
+
+		specs = append(specs, spec)
+	}
+
+	return specs, nil
+}
+
+// getHelmIndex returns an helm repository IndexFile object
```

Comment:
typo: an helm repository to a helm repository.

I realize there is some debate on whether a vs an in front of h depending on whether the h is silent (as in hour) or pronounced as in hello. I think 'a helm repository' sounds more correct.